src/components/Categories/Categories.tsx

```diff
@@ -109,7 +119,40 @@ const Categories: FC = () => {
   }, [input]);
 
   useEffect(() => {
-    dropDownValue && filterReset();
+    dropDownValue.length && filterReset();
+    category.length &&
+      brand.length &&
+      dispatch(
+        addFilterValues({
+          category: [category[0]],
+          brand: [brand[0]],
+          rating: [],
+          priceMin: Math.floor(price[0]),
+          priceMax: Math.ceil(price[1]),
+        })
+      );
+    setInput({
+      ...input,
+      priceMin: Math.floor(price[0]),
+      priceMax: Math.ceil(price[1]),
+    });
+
+    const categoryIndex = categoryValues
+      .map((el, id) => (el.value.toLocaleLowerCase() === String(category[0]).toLocaleLowerCase() ? id : el))
+      .filter((el) => typeof el === "number")
```

Comment:
Think you can try .filter(Number) or .filter(Number.isInteger)

Reply:
Good idea , added